db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -3471,20 +3471,21 @@ public void clearSPSPlans() throws StandardException{
 
         FormatableBitSet columnToReadSet=new FormatableBitSet(SYSSTATEMENTSRowFactory.SYSSTATEMENTS_COLUMN_COUNT);
         FormatableBitSet columnToUpdateSet=new FormatableBitSet(SYSSTATEMENTSRowFactory.SYSSTATEMENTS_COLUMN_COUNT);
-        columnToUpdateSet.set(SYSSTATEMENTSRowFactory.SYSSTATEMENTS_VALID-1);
-        columnToUpdateSet.set(SYSSTATEMENTSRowFactory.SYSSTATEMENTS_CONSTANTSTATE-1);
-
+        for(int i=0;i<SYSSTATEMENTSRowFactory.SYSSTATEMENTS_COLUMN_COUNT;i++){
+            if (i+1==SYSSTATEMENTSRowFactory.SYSSTATEMENTS_CONSTANTSTATE) {
+                columnToUpdateSet.set(i);
```

Comment:
why is `columnToUpdateSet.set(i);` done in if and in else?
shouldn't it rather be

```
       for(int i=0;i<SYSSTATEMENTSRowFactory.SYSSTATEMENTS_COLUMN_COUNT;i++){
            if (i+1 != SYSSTATEMENTSRowFactory.SYSSTATEMENTS_CONSTANTSTATE) {
                columnToReadSet.set(i);
            }
            columnToUpdateSet.set(i);
        }
```

Reply:
Done!